external/loaders/loaders/mappers/_fine_resolution_budget.py

```diff
@@ -155,14 +167,18 @@ def __init__(self, url):
         if len(self.files) == 0:
             raise ValueError("No file detected")
 
-    def _parse_file(self, url):
+    def _parse_file(self, url) -> Tuple[str, int]:
         pattern = r"tile(.)\.nc"
         match = re.search(pattern, url)
+        if match is None:
+            raise ValueError(
+                f"invalid url, must contain pattern {pattern} but received {url}"
+            )
         date = vcm.parse_timestep_str_from_path(url)
         tile = match.group(1)
         return date, int(tile)
 
-    def __getitem__(self, key: str) -> xr.Dataset:
+    def __getitem__(self, key):
```

Comment:
why do we need to remove the type hints here?

Reply:
These are already type hinted (according to mypy) due to the inheritance from `Mapping`. When I added manual type hints to some routines (but probably not this one), I got issues with things like mypy saying AbstractSet is incompatible with Iterable, so I just removed the method-level type hints.